apps/lib/apps.c

```diff
@@ -1718,6 +1718,10 @@ CA_DB *load_index(const char *dbfile, DB_ATTR *db_attr)
     }
 
     retdb->dbfname = OPENSSL_strdup(dbfile);
+    if (retdb->dbfname == NULL){
+        ERR_raise_data(ERR_LIB_SYS, errno, "Out of memory while copying filename: %s", dbfile);
+        goto err;
```

Comment:
This still seems to return retdb, but then only partially set correctly. I think we want to free it and set it to NULL in this case.

Reply:
I think the style guide has a space between the `)` and `{`

Reply:
Sorry for the style mistake. I fixed it in my new commit.

Besides free `retdb`, I also freed `retdb->db` because I see the origin error processing (`err:`) tries to free `tmpdb`.